spikesorters/yass/yass.py

```diff
@@ -0,0 +1,280 @@
+import copy
+from pathlib import Path
+import os
+import numpy as np
+from numpy.lib.format import open_memmap
+import sys
+import yaml
+
+import spikeextractors as se
+
+from ..basesorter import BaseSorter
+from ..sorter_tools import recover_recording
+
+from ..utils.shellscript import ShellScript
+
+
+from shellscript import ShellScript
+
+try:
+    import yass
+    HAVE_YASS = True
+except ImportError:
+    HAVE_YASS = False
+
+print ("HAVE_YASS: ", HAVE_YASS) 
+
+#####################################################################
+############# YASSSORTER CLASS ######################################
+#####################################################################
+class YassSorter(BaseSorter):
+    """ 
+    """
+
+    sorter_name = 'yass'
+    requires_locations = False
+    verbose = False
+    is_installed = HAVE_YASS
+
+    # _default_params comes from default Yass config.yaml file; 
+    #  Should be autoloaded from yass_installation_directory/samples/10chan/config.yaml
+    #  - for now a copy is saved here: spikesorters/spikesorters/yass/config.yaml
+    
+#     _default_params = {
+#         'adjacency_radius': 100,  # Channel neighborhood adjacency radius corresponding to geom file
+#         'filter': True,
+#         }
+
+    # ALESSIO: Move thsi file locally to the yass directory;
+    default_config_fname = './config_sample.yaml'
+    with open(default_config_fname) as f:
+        _default_params = yaml.load(f, Loader=yaml.FullLoader)   
+    
+    _params_description = {
+        'data': " ...",
+    }
+
+    # 
+    sorter_description = """Yass uses Neural Networks and SuperResolution Deconvolution to sort Retinal and 
+                        cortical data. 
+                        For more information see https://www.biorxiv.org/content/10.1101/2020.03.18.997924v1
+                        """
+
+    installation_mesg = """\nTo Install and Use Yass follow the wiki: 
+                            https://github.com/paninski-lab/yass/wiki
+                        """
+
+    def __init__(self, **kargs):
+        print (**kargs)
+        #BaseSorter.__init__(self, rec, output_folder)
+        self.params = self._default_params
```

Comment:
You must do a copy here because `_default_params` is class attr and params is instance attribute.